pkg/infrastructure/azure/compute.go

```diff
@@ -0,0 +1,221 @@
+	"github.com/sirupsen/logrus"
+)
+
+// CreateImageGalleryInput contains the input paramters for creating a image
+// gallery.
+type CreateImageGalleryInput struct {
+	SubscriptionID     string
+	ResourceGroupName  string
+	GalleryName        string
+	Region             string
+	Tags               map[string]*string
+	TokenCredential    azcore.TokenCredential
+	CloudConfiguration cloud.Configuration
+}
+
+// CreateImageGalleryOutput contains the return values after creating a image
+// gallery.
+type CreateImageGalleryOutput struct {
+	ComputeClientFactory *armcompute.ClientFactory
+	Gallery              *armcompute.Gallery
+}
+
+// CreateImageGallery creates a image gallery
+func CreateImageGallery(ctx context.Context, in *CreateImageGalleryInput) (*CreateImageGalleryOutput, error) {
+	computeClientFactory, err := armcompute.NewClientFactory(
+		in.SubscriptionID,
+		in.TokenCredential,
+		&arm.ClientOptions{
+			ClientOptions: policy.ClientOptions{
+				Cloud: in.CloudConfiguration,
+			},
+		},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get compute client factory: %w", err)
+	}
+
+	galleriesClient := computeClientFactory.NewGalleriesClient()
+
+	galleriesPoller, err := galleriesClient.BeginCreateOrUpdate(
+		ctx,
+		in.ResourceGroupName,
+		in.GalleryName,
+		armcompute.Gallery{
+			Location:   to.Ptr(in.Region),
+			Properties: &armcompute.GalleryProperties{
+				// Fill this in
+			},
+			Tags: in.Tags,
+		},
+		nil,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create gallery %s: %w", in.GalleryName, err)
+	}
+
+	galleriesPollDone, err := galleriesPoller.PollUntilDone(ctx, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to finish creating gallery %s: %w", in.GalleryName, err)
+	}
+
+	return &CreateImageGalleryOutput{
+		ComputeClientFactory: computeClientFactory,
+		Gallery:              to.Ptr(galleriesPollDone.Gallery),
+	}, nil
+}
+
+// CreateGalleryImageInput contains the input paramters for creating a gallery
+// image.
+type CreateGalleryImageInput struct {
+	ResourceGroupName    string
+	GalleryName          string
+	GalleryImageName     string
+	Region               string
+	Tags                 map[string]*string
+	TokenCredential      azcore.TokenCredential
+	CloudConfiguration   cloud.Configuration
+	OSType               armcompute.OperatingSystemTypes
+	OSState              armcompute.OperatingSystemStateTypes
+	HyperVGeneration     armcompute.HyperVGeneration
+	Publisher            string
+	Offer                string
+	SKU                  string
+	ComputeClientFactory *armcompute.ClientFactory
+}
+
+// CreateGalleryImageOutput contains the return values after creating a gallery
+// image.
+type CreateGalleryImageOutput struct {
+	GalleryImage *armcompute.GalleryImage
+}
+
+// CreateGalleryImage creates a gallery image
+func CreateGalleryImage(ctx context.Context, in *CreateGalleryImageInput) (*CreateGalleryImageOutput, error) {
```

Comment:
As these functions can take a while, it would be good to have info level debugging statements at the beginning and ends

Reply:
I've added debug logging